geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GenerateFeatureWrappers.scala

```diff
@@ -0,0 +1,185 @@
+    val majorBinding = classToString(Some(ad.getType.getBinding))
+    val binding = if (ad.isCollection) {
+      val subtype = classToString(SimpleFeatureTypes.getCollectionType(ad))
+      s"$majorBinding[$subtype]"
+    } else if (ad.isMap) {
+      val keyType = classToString(Option(ad.getUserData.get("keyclass").asInstanceOf[Class[_]]))
+      val valueType = classToString(Option(ad.getUserData.get("valueclass").asInstanceOf[Class[_]]))
+      s"$majorBinding[$keyType,$valueType]"
+    } else {
+      majorBinding
+    }
+    AttributeDetails(ad.getLocalName, sft.indexOf(ad.getLocalName), binding)
+  }
+
+  private def classToString(clas: Option[Class[_]]) = clas.map(_.getCanonicalName).getOrElse("String")
+}
+
+object GenerateFeatureWrappers {
+
+  val className = "SimpleFeatureWrappers"
+
+  /**
+   * Builds all implicit classes in a wrapper object
+   *
+   * @param sfts
+   * @param pkg
+   * @return
+   */
+  def buildAllClasses(sfts: Seq[SimpleFeatureType], pkg: String): String = {
+    val sb = new StringBuilder()
+    sb.append(s"package $pkg\n\n")
+    sb.append(s"object $className {")
+    sfts.foreach(sft => sb.append("\n\n").append(buildClass(sft, "  ")))
+    sb.append("\n}")
+    sb.toString()
+  }
+
+  /**
+   * Builds a single implicit class
+   *
+   * @param sft
+   * @param tab
+   * @return
+   */
+  def buildClass(sft: SimpleFeatureType, tab: String): String = {
+    import scala.collection.JavaConversions._
+
+    val attrs = sft.getAttributeDescriptors.map(AttributeDetails(_, sft))
+
+    val sb = new StringBuilder()
+    sb.append(s"${tab}implicit class ${sft.getTypeName}")
+    sb.append("(val sf: org.opengis.feature.simple.SimpleFeature) extends AnyVal {\n")
+    attrs.foreach { a =>
+      sb.append("\n")
+      sb.append(s"$tab  ${a.getter}\n")
+      sb.append(s"$tab  ${a.optionGetter}\n")
+      sb.append(s"$tab  ${a.setter}\n")
+    }
+    sb.append(
+      s"""
+        |$tab  def debug(): String = {
+        |$tab    import scala.collection.JavaConversions._
+        |$tab    val sb = new StringBuilder(s"$${sf.getType.getTypeName}:$${sf.getID}")
+        |$tab    sf.getProperties.foreach(p => sb.append(s"|$${p.getName.getLocalPart}=$${p.getValue}"))
+        |$tab    sb.toString()
+        |$tab  }
+        |""".stripMargin)
+    sb.append(s"$tab}")
+
+    sb.toString()
+  }
+
+  /**
+   * Recursively looks for configuration files of the pattern 'format-*.conf'
+   *
+   * @param file
+   * @return
+   */
+  def findFormatFiles(file: File): Seq[File] = {
+    if (!file.isDirectory) {
+      val name = file.getName
+      if (name.startsWith("format-") && name.endsWith(".conf")) {
+        Seq(file)
+      } else {
+        Seq.empty
+      }
+    } else {
+      file.listFiles().flatMap(findFormatFiles)
+    }
+  }
+
+  /**
+   * Creates implicit wrappers for any typesafe config format files found under src/main/resources
+   *
+   * @param args (0) - base directory for the maven project
```

Comment:
Not for this commit, but at some point we ought to make this a full blown maven plugin.  We can keep the main method around for one-off use cases but it can just wrap the maven plugin.